test/model/autocomplete_test.dart

```diff
@@ -0,0 +1,36 @@
+import 'package:checks/checks.dart';
+import 'package:test/scaffolding.dart';
+import 'package:zulip/api/model/model.dart';
+import 'package:zulip/model/autocomplete.dart';
+
+import '../example_data.dart' as eg;
+
+void main() {
+  test('MentionAutocompleteQuery.testUser', () {
+    doCheck(String rawQuery, User user, bool expected) {
+      final result = MentionAutocompleteQuery(rawQuery).testUser(user);
+      expected ? check(result).isTrue() : check(result).isFalse();
+    }
+
+    doCheck('', eg.user(fullName: 'Full Name'), true);
```

Comment:
Also include tests with empty `fullName` ­— one where query is also empty, one where nonempty.

That isn't a super likely case, but it wouldn't be hard to have a bug where the search crashed on it.